### PR TITLE
ci: Push text-gen-server release images to Quay.io

### PR DESCRIPTION
**Pull request by ckadner**
#### Motivation

Server release images need a container registry home.

#### Modifications

Push to quay.io instead of ghcr.io

#### Result

Images pushed to quay.io

https://quay.io/repository/wxpe/text-gen-server?tab=tags

#### Related Issues

N/A

@joerunde

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -78,12 +84,25 @@ jobs:
...
+        uses: docker/metadata-action@v5
+        with:
+          images: |
+            ${{ env.SERVER_IMAGE }}
+          tags: |
+            type=ref,event=branch
+            type=semver,pattern={{version}}
+            type=semver,pattern={{major}}.{{minor}}
+            type=sha,enable=true,priority=100,prefix=,suffix=,format=short     
+            type=sha,enable=true,priority=100,prefix=${{ github.ref_name }}.,suffix=,format=short
```
Comment (ckadner) [inline .github/workflows/build.yml:98]: @joerunde -- I replicated what you did in https://github.com/IBM/text-generation-router/blob/main/scripts/get_image_tags.sh

Comment (joerunde) [reply .github/workflows/build.yml:98]: nice! I'll have to go backport

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -78,12 +84,25 @@ jobs:
...
+            type=sha,enable=true,priority=100,prefix=,suffix=,format=short     
+            type=sha,enable=true,priority=100,prefix=${{ github.ref_name }}.,suffix=,format=short     
+
       - name: "Docker build server-release"
         uses: docker/build-push-action@v5
         with:
           context: .
           target: server-release
-          tags: ${{ env.SERVER_IMAGE }}
+          tags: ${{ steps.meta.outputs.tags }}
```
Comment (joerunde) [inline .github/workflows/build.yml:105]: so clean

**Review by joerunde — APPROVED** [review]
let's get this merged!

**Pull request merged**